### PR TITLE
Adding react-week4 homework

### PR DESCRIPTION
**Pull request by priyankapandey13**

### CODE REVIEW

File: react/week4/managing-state/src/TodoListItems.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const details = <div>
+                    <span>{props.todoitem.login}</span>  
+                    <img src={props.todoitem.avatar_url} alt={props.todoitem.avatar_url}></img>
+                  </div>
+  return <li>
+            <div>
+              {details}
+            </div>
+          </li>;
+}
```
Comment (braedongough) [inline react/week4/managing-state/src/TodoListItems.js]: I think storing half of your component in a variable is a bit of an unnecessary abstraction. I always try to keep things as simple and readable as possible: 
```suggestion
const TodoListItems=(props) => {    
  return <li>
            <div>
                    <span>{props.todoitem.login}</span>  
                    <img src={props.todoitem.avatar_url} alt={props.todoitem.avatar_url}></img>
            </div>
          </li>;
}
```

File: react/week4/managing-state/src/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+// Importing components
+import Context from './context';
+import Searchuser from './searchuser';
+import ListItems from './userListItems';
+
+// Context
+const Contextprovider = Context.Provider;
+
+function FetchArray(event){
```
Comment (braedongough) [inline react/week4/managing-state/src/index.js]: You should only uppercase Component names. Functions and variables should still be camel case: 
```suggestion
function fetchArray(event){
```

It's necessary for react that components are uppercased: https://reactjs.org/docs/jsx-in-depth.html#specifying-the-react-element-type

File: react/week4/managing-state/src/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  fetchlist=(event) => {    
+    FetchArray(event).then(data => {     
+      this.setState({ Todo: data.items})
+    }).catch((error) => {
+      if(error){
+      alert(`This Process Failed : ${error}`);
+      return
+      }
+    }).finally(() => {
```
Comment (braedongough) [inline react/week4/managing-state/src/index.js:33]: nice use of `finally`!

File: react/week4/managing-state/src/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      this.setState({isLoading:false})
+    })
+  }
+
+  handleChange=(event) => {
+    const inputevent = event.target.value;
+    if(!inputevent){
+      return
+    }
+    this.setState({isLoading:true})
```
Comment (braedongough) [inline react/week4/managing-state/src/index.js:43]: I would move this inside the `fetchlist` method instead. This way you can always be guaranteed the state will be changed as you expect. In this setup, if somewhere were to call handleChange without fetchlist, you'd be stuck loading forever.

File: react/week4/managing-state/src/index.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+// Context
+const Contextprovider = Context.Provider;
+
+function FetchArray(event){
+  const URL=`https://api.github.com/search/users?q=${event}`;
+    return fetch(URL).then(response => response.json())        
+}
+
+class TodoList extends React.Component{
```
Comment (braedongough) [inline react/week4/managing-state/src/index.js]: You're not using this component to store todos, it's generating a list of users from github. I would rename this component and all uses of the word todo to correctly reflect what you're actually rendering in this component.

**Review by braedongough — APPROVED** [review]
Good work!

File: react/week4/managing-state/src/TodoListItems.js (hunk 1 above)
Comment (priyankapandey13) [reply react/week4/managing-state/src/TodoListItems.js]: Yes, make sense

File: react/week4/managing-state/src/index.js (hunk 5 above)
Comment (priyankapandey13) [reply react/week4/managing-state/src/index.js]: I used few things from my old homework that's why the name was "Todo", but i have changed it in every file. Will keep it in mind for future

**Pull request merged**